### ISSUE TITLE
[Workspace block aws_s3_bucket_versioning]

### ISSUE THREAD

**Issue by lucasp0r** [hashicorp/terraform-provider-aws#23510]
Hello Everyone,

After a few hours to understand how I got an error in this resource, I understood what happens. 

If I use my workspace and I got lock-in that workspace, if unblock that, and I try to apply or destroy these resources, I got an error.

So I tried to simulate this in others services like EKS, ECS, ALB, ROUTE53 etc.. and more of that. Same process, lock-in in my workspace, unblock that, and apply/destroy, but works properly in that resources. So I just got a problem with **aws_s3_bucket_versioning** 

Is a kind of strange like this is a new resource in provider 4.0.0 +, maybe a bug? Someone can help? 

***Code***

```
resource "aws_s3_bucket" "bucket" {
  bucket = var.name_bucket
}

resource "aws_s3_bucket_acl" "bucket" {
  bucket = aws_s3_bucket.bucket.id
  acl    = "private"
}

resource "aws_s3_bucket_versioning" "versioning_bucket" {
  bucket = aws_s3_bucket.bucket.id
  versioning_configuration {
    status = "Enabled"
  }
}
```

***Error***

```

│ Error: error waiting for S3 Bucket Versioning status for bucket (name_bucket): couldn't find resource (4 retries)
│ 
│   with module.s3.aws_s3_bucket_versioning.versioning_bucket,
│   on modules/s3/main.tf line 10, in resource "aws_s3_bucket_versioning" "versioning_bucket":
│   10: resource "aws_s3_bucket_versioning" "versioning_bucket" {
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Latest version for both

### Affected Resource(s)

* aws_s3_bucket_versioning

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Steps to Reproduce

1. `terraform apply/destroy` 

### References

* #0000

**Comment by justinretzolk** [comment]
Hey @lucasp0r 👋 Thank you for taking the time to raise this! So that we have all of the necessary information in order to look into this, can you update the issue description to include the rest of the information requested in the template (particularly debug logs, redacted as necessary)? 

I also attempted to reproduce this with the Terraform configuration supplied and was not able to reproduce the error; can you expand a bit more on the steps to reproduce?

**Comment by kevinscholz** [comment]
I received the same error. I also use workspaces and the error occurred after the S3 bucket was deleted outside of Terraform. During resource refresh I also got `Error: error waiting for S3 Bucket Versioning status for bucket (my-bucket): couldn't find resource (4 retries)`. I saw some similar errors with other resources in the past, when the refresh did not handle the absence of the resource correctly.

**Comment by kevinli92924** [comment]
I was having the same error. This is probably because you deleted your resource manually. In this case, you probably deleted the S3 bucket that was created by Terraform. Try manually add an S3 bucket with the same bucket name and redo it. See how it goes.

**Comment by lerrigatto** [comment]
Hi! I am having the same issue when applying from scratch. If I run an apply targeting only the bucket and then re-apply everything, it works.

**Comment by erikpaasonen** [comment]
Pretty sure this behavior is due to the fact that the code behind `aws_s3_bucket_versioning` currently has a short-circuit error check whose effect is to bypass the error handling code. The unreachable code would gracefully remove this resource from the tfstate file if the bucket were to be deleted outside of Terraform. most other `aws_s3_bucket` secondary resource objects have this check, just not this one. (Also found it missing from the `aws_s3_bucket_policy` resource as well, so included adding it there too.) opened #24399 with a proposed fix.

**Comment by ingletkimi** [comment]
Manually adding the s3 bucket with versioning enable works, yes. I see this is closed now but I'm still getting the same error. Is there an open discussion for aws_s3_bucket_versioning status?

**Comment by anGie44** [comment]
HI @ingletkimi , the fix hasn't been released just yet, will likely be available later today with `v4.12.0` of the AWS provider. Are you getting the error with a local build of the provider? 

> Is there an open discussion for aws_s3_bucket_versioning status?

Not that I am aware of. There is this issue https://github.com/hashicorp/terraform-provider-aws/issues/23809 that perhaps is related but if you have a separate feature request or bug you'd like add, feel free to create a new issue in the repo.

**Comment by ProgrammerCats** [comment]
I ran into the same issue, but in my case I was able to avoid the error by doing the following to exclude it from local resource management.

terraform state rm aws_s3_bucket.**********
terraform state rm aws_s3_bucket_lifecycle_configuration.**********
terraform state rm aws_s3_bucket_policy.**********
terraform state rm aws_s3_bucket_versioning.**********
terraform state rm aws_s3_bucket_acl.**********

********** is own resource name.
